django/contrib/auth/models.py

```diff
@@ -76,7 +76,18 @@ class Meta:
         ordering = ["content_type__app_label", "content_type__model", "codename"]
 
     def __str__(self):
-        return "%s | %s" % (self.content_type, self.name)
+        names = { "content_type": self.content_type }
+
+        if "Can add" in self.name:
```

Comment:
Would `startswith` be a more accurate check?

```suggestion
        if self.name.startswith("Can add "):
```

Reply:
Definitely. And the fallback after those conditionals should probably be `return "%s | %s" % (self.content_type, _(self.name))`, so custom permissions are translate-able (would additionally require people to hard-code a copy of their permission names somewhere in their code for `makemessages` to pick up).

Reply:
Ah cool 👍